pyzabbix/api.py

```diff
@@ -255,7 +255,7 @@ def do_request(self, method, params=None):
             err.update({'json': str(request_json)})
             msg_str = "Error {code}: {message}, {data} while sending {json}"
             msg = msg_str.format(**err)
-            raise ZabbixAPIException(msg, err['code'])
+            raise ZabbixAPIException(msg, err['code'], err)
```

Comment:
I like overall idea, but I have two conerns:
* Backward compatibility, since many people already using current syntax and we definitely don't wanna break it. So we have to think how it support both cases.
* We may want just do  `raise ZabbixAPIException(err)` and extend `ZabbixAPIException` class to handle message format.

Reply:
>Backward compatibility, since many people already using current syntax and we definitely don't wanna break it. So we have to think how to support both cases.

Thats why I appended `err` to `(msg, err['code'])` to avoid backward compatibility problems. 

>We may want just do raise ZabbixAPIException(err) and extend ZabbixAPIException class to handle message format.

I agree, but I decided not to do that as `ZabbixAPIException(err)` to avoid backward compatibility problems, see below. Sorry, no clue at this point how to extend class to handle both `ZabbixAPIException(err)` and `ZabbixAPIException(msg, err['code'])`

Reply:
How do you see it?